infra/modules/service/scheduled_jobs_workflow_role.tf

```diff
@@ -0,0 +1,108 @@
+# This role and policy are used by the Step Functions state machine that manages the scheduled jobs workflow.
+
+resource "aws_iam_role" "scheduled_jobs_workflow_role" {
+  name               = "${var.service_name}-scheduled-jobs-workflow"
+  assume_role_policy = data.aws_iam_policy_document.scheduled_jobs_workflow_assume_role_policy.json
+}
+
+resource "aws_iam_policy" "scheduled_jobs_workflow_policy" {
+  name   = "${var.service_name}-scheduled-jobs-workflow"
+  policy = data.aws_iam_policy_document.scheduled_jobs_workflow_policy.json
+}
+
+resource "aws_iam_role_policy_attachment" "scheduled_jobs_workflow_policy_attachment" {
+  role       = aws_iam_role.scheduled_jobs_workflow_role.name
+  policy_arn = aws_iam_policy.scheduled_jobs_workflow_policy.arn
+}
+
+# policy sourced via: https://docs.aws.amazon.com/step-functions/latest/dg/procedure-create-iam-role.html
+data "aws_iam_policy_document" "scheduled_jobs_workflow_assume_role_policy" {
+  statement {
+    sid = "ECSTasksAssumeRole"
+    actions = [
+      "sts:AssumeRole"
+    ]
+    principals {
+      type = "Service"
+      identifiers = [
+        "states.amazonaws.com",
+      ]
+    }
+    condition {
+      test     = "ArnLike"
+      variable = "aws:SourceArn"
+      values = [
+        "arn:aws:states:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:stateMachine:*"
+      ]
+    }
+
+    condition {
+      test     = "StringLike"
+      variable = "aws:SourceAccount"
+      values = [
+        data.aws_caller_identity.current.account_id
+      ]
+    }
+  }
+}
+
+#tfsec:ignore:aws-iam-no-policy-wildcards
+data "aws_iam_policy_document" "scheduled_jobs_workflow_policy" {
+  # checkov:skip=CKV_AWS_111:These permissions are scoped just fine
+
+  # policy sourced via: https://docs.aws.amazon.com/step-functions/latest/dg/cw-logs.html
+  statement {
+    sid = "UnscopeLogsPermissions"
+    actions = [
+      "logs:CreateLogDelivery",
+      "logs:CreateLogStream",
+      "logs:GetLogDelivery",
+      "logs:UpdateLogDelivery",
+      "logs:DeleteLogDelivery",
+      "logs:ListLogDeliveries",
+      "logs:PutLogEvents",
+      "logs:PutResourcePolicy",
+      "logs:DescribeResourcePolicies",
+      "logs:DescribeLogGroups",
+    ]
+    resources = ["*"]
+  }
+
+  # policy sourced via: https://docs.aws.amazon.com/step-functions/latest/dg/ecs-iam.html
+  statement {
+    sid = "StepFunctionsEvents"
+    actions = [
+      "events:PutTargets",
+      "events:PutRule",
+      "events:DescribeRule",
+    ]
+    resources = [
+      "arn:aws:events:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:rule/StepFunctionsGetEventsForECSTaskRule",
+    ]
+  }
+
+  # policy sourced via: https://docs.aws.amazon.com/step-functions/latest/dg/ecs-iam.html
+  statement {
+    sid = "StepFunctionsRunTask"
+    actions = [
+      "ecs:RunTask",
+      "ecs:StopTask",
+      "ecs:DescribeTasks",
+    ]
+    resources = ["*"]
```

Comment:
Unfortunately not, because this is working on tasks, where the tasks aren't defined at build time. This in contrast to the cluster, which is defined at build time. Also the arn for the tasks is a UUID. At best, we could do something like:

```
condition
  type = StringLike
  key = tags!ecs-cluster
  value = aws_ecs_cluster.cluster.name
```

Which would of course require us to tag our ECS tasks.

So its possible to scope this down further, but I wouldn't really recommend it.

Reply:
Hmm I'm skeptical. If you look at events_role.tf it's working on tasks:
```
data "aws_iam_policy_document" "run_task" {
  statement {
    effect    = "Allow"
    actions   = ["ecs:RunTask"]
    resources = ["${aws_ecs_task_definition.app.arn_without_revision}:*"]
    condition {
      test     = "ArnLike"
      variable = "ecs:cluster"
      values   = [aws_ecs_cluster.cluster.arn]
    }
  }
```
Seems like if there are multiple apps, the workflow orchestrator for one app shouldn't be able to run tasks for the other app